geomstats/information_geometry/gamma.py

```diff
@@ -0,0 +1,1200 @@
+
+            if curve.min() < 0:
+                return np.inf, np.inf, curve, np.nan
+
+            velocity_sqnorm = self.squared_norm(vector=velocity, base_point=curve)
+            length = gs.sum(velocity_sqnorm ** (1 / 2)) / n_times
+            energy = gs.sum(velocity_sqnorm) / n_times
+            return energy, length, curve, velocity
+
+        def cost_jacobian(param):
+            """Compute the jacobian of the cost function at polynomial curve.
+
+            Parameters
+            ----------
+            param : array-like, shape=(degree - 1, dim)
+                Parameters of the curve coordinates' polynomial functions of time.
+
+            Returns
+            -------
+            jac : array-like, shape=(dim * (degree - 1),)
+                Jacobian of the cost function at polynomial curve.
+            """
+            last_coef = end_point - initial_point - gs.sum(param, axis=0)
+            coef = gs.vstack((initial_point, param, last_coef))
+
+            t = gs.linspace(0.0, 1.0, n_times)
+            t_position = [t**i for i in range(degree + 1)]
+            t_position = gs.stack(t_position)
+            position = gs.einsum("ij,ik->kj", coef, t_position)
+
+            t_velocity = [i * t ** (i - 1) for i in range(1, degree + 1)]
+            t_velocity = gs.stack(t_velocity)
+            velocity = gs.einsum("ij,ik->kj", coef[1:], t_velocity)
+
+            kappa, gamma = position[:, 0], position[:, 1]
+            kappa_dot, gamma_dot = velocity[:, 0], velocity[:, 1]
+
+            jac_kappa_0 = (
+                (gs.polygamma(2, kappa) + 1 / kappa**2) * kappa_dot
+                + gamma_dot**2 / gamma
+            ) * t_position[1:-1]
+            jac_kappa_1 = (2 * gs.polygamma(1, kappa) * kappa_dot) * t_velocity[:-1]
+
+            jac_kappa = jac_kappa_0 + jac_kappa_1
+
+            jac_gamma_0 = (-kappa * gamma_dot**2 / gamma**2) * t_position[1:-1]
+            jac_gamma_1 = (2 * kappa * gamma_dot / gamma) * t_velocity[:-1]
+
+            jac_gamma = jac_gamma_0 + jac_gamma_1
+
+            jac = gs.vstack([jac_kappa, jac_gamma])
+
+            cost_jac = gs.sum(jac, axis=1)
+            return cost_jac
+
+        def f2minimize(x):
+            """Compute function to minimize."""
+            param = gs.transpose(x.reshape((dim, degree - 1)))
+            res = cost_fun(param)
+            return res[0]
+
+        def jacobian(x):
+            """Compute jacobian of the function to minimize."""
+            param = gs.transpose(x.reshape((dim, degree - 1)))
+            return cost_jacobian(param)
+
+        dim = initial_point.shape[0]
+        x0 = gs.ones(dim * (degree - 1))
+        jac = jacobian if jac_on else None
+        sol = minimize(f2minimize, x0, method=method, jac=jac)
+        opt_param = sol.x.reshape((dim, degree - 1)).T
+        _, dist, curve, velocity = cost_fun(opt_param)
+
+        return dist, curve, velocity
+
+    def _geodesic_bvp(
```

Comment:
This function should be tested, see e.g. [here](https://github.com/geomstats/geomstats/blob/9bad7b8cc618fc0b9cc3dee13d01698a8f86bb11/tests/tests_geomstats/test_dirichlet.py#L167).